src/apps/fuzzers/fuzzerCompact.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file
+ * @brief Fuzzer program for compactCells and uncompactCells
+ */
+
+#include "aflHarness.h"
+#include "h3api.h"
+#include "utility.h"
+
+const int MAX_UNCOMPACT_RES = 9;
+const int64_t MAX_UNCOMPACT_SIZE = 4000000;
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+    if (size < sizeof(H3Index) * 2) {
+        return 0;
+    }
+
+    uint8_t res = *data;
+    uint8_t uncompactRes = (*(data + 1)) % (MAX_UNCOMPACT_RES + 1);
```

Comment:
It would be nice if we could try uncompacting to res 15 (or beyond, and catch the error), as I'd expect that would be the place where out of bounds issues would come up. Maybe we put a bound on the uncompact res delta instead of res = 9?

Reply:
I'd suggest deferring this to a follow up PR. There are some other enhancements and fixes to these functions and this fuzzer but I would prefer to make them a new, more digestible PR.